src/validationinterface.cpp

```diff
@@ -37,6 +37,8 @@ struct MainSignalsInstance {
 };
 
 static CMainSignals g_signals;
+static std::unordered_map<CTxMemPool*, boost::signals2::connection> g_connNotifyEntryRemoved;
+static std::unordered_map<CValidationInterface*, std::array<boost::signals2::connection, 9>> g_connMainSignals;
```

Comment:
It seems like these two maps would most naturally be `MainSignalsInstance` members. I don't think it is good to arbitrary divide the validationinterface state up between `g_signals->m_internals` and these new global variables (unless there's a specific reason for doing this that I'm not seeing).

Reply:
I was trying to do this however MainSignalsInstance is not created when `RegisterWithMempoolSignals()` called. So I only move `g_connMainSignals` into a member of `MainSignalsInstance`. Keeping `g_connNotifyEntryRemoved` as a static global variable.